src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -956,3 +943,98 @@ def assign_last_name_ids(self, pop: pd.DataFrame) -> pd.Series:
+            population["household_id"].isin(household_ids_without_reference_person)
+        ]
+
+        # Find oldest member in each household and make them new reference person
+        # This is a series with household_id as the index and the new reference person as the value
+        new_reference_persons = (
+            population.loc[households_to_update_idx].groupby(["household_id"])["age"].idxmax()
+        )
+        # Preserve old relationship of new reference person before assigning them as new reference persons
+        new_reference_persons_relationship_to_old_reference_person = population.loc[
+            new_reference_persons, ["household_id", "relation_to_household_head"]
+        ]
+        population.loc[
+            new_reference_persons, "relation_to_household_head"
+        ] = "Reference person"
+
+        # Update simulants born in simulation as biological children to their mother if mother is new reference person
+        biological_children_idx = population.index[
+            (population["date_of_birth"] > self.start_time)
+            & (
+                population["household_id"].isin(
+                    new_reference_persons_relationship_to_old_reference_person["household_id"]
+                )
+            )
+            & (
+                population["guardian_1"]
+                == population["household_id"].map(
+                    new_reference_persons_relationship_to_old_reference_person.reset_index().set_index(
+                        "household_id"
+                    )[
+                        "index"
+                    ]
+                )
+            )
+        ]
+        population.loc[
+            biological_children_idx, "relation_to_household_head"
+        ] = "Biological child"
+
+        # Update other household members relationship to new reference person
+        for relationship in new_reference_persons_relationship_to_old_reference_person[
+            "relation_to_household_head"
+        ].unique():
+            relationship_map = self.reference_person_update_relationships_map.loc[
+                self.reference_person_update_relationships_map[
+                    "new_reference_person_relationship_to_old_reference_person"
+                ]
+                == relationship
+            ].set_index("relationship_to_old_reference_person")
+
+            household_ids_to_update = (
+                new_reference_persons_relationship_to_old_reference_person.loc[
+                    new_reference_persons_relationship_to_old_reference_person[
+                        "relation_to_household_head"
+                    ]
+                    == relationship,
+                    "household_id",
+                ]
+            )
+            simulants_to_update_idx = (
+                population.index[population["household_id"].isin(household_ids_to_update)]
+                .difference(new_reference_persons_relationship_to_old_reference_person.index)
```

Comment:
How does this differ from the `new_reference_persons` variable?

Reply:
But can you not just do `.difference(new_reference_persons)` (note how there is no `.index`)? That surprises me, since above you use new_reference_persons to `.loc`.

Reply:
It has to be on the index.  You can do .loc on values or an index.  This also surprised me.

Reply:
The `new_reference_persons` variable is only ever used in this way (as an index) -- so, when you define it, you can just do `pd.Index(...)` and then a) it's easier to understand the `.loc` and b) you can use it here.

Reply:
New reference person is used as an index by looking up the values of that series in the population index.  I see what you are saying with pd.Index(new_reference_persons) and this would work but has also been done already when creating the new_reference_person_prev_relationship variable.  We have to create that variable anyways to store the old relationships so it is not doing any extra work here.